Knowledge_Check_Wk08/client/src/App.js

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react';
+import "./App.css";
+
+export default function App() {
+
+	const [currentQuestion, setCurrentQuestion] = useState(0);
+
+	const [showScore, setShowScore] = useState(false);
+
+	const [score, setScore] = useState(0);
+
+	const [questions, setQuestions] = useState([]);
+
+	const [resetGame, setResetGame] = useState(false);
+
+	const handleAnswerButtonClick = (choice) => {
+		if (choice === questions[currentQuestion].correct) {
+			alert("the answer is correct!") 
+			setScore(score + 1);      
+		}
+		const nextQuestion = currentQuestion + 1;
+		if (nextQuestion < questions.length) {
+			setCurrentQuestion(nextQuestion);
+		} else {
+			setShowScore(true);
+		}
+	};
+
+	const handleResetGameButton = () => {
+		setResetGame(true);
+	}
+
+	useEffect(() => {
+	    if (resetGame){
+		setShowScore(false);
+		setScore(0);
+		setCurrentQuestion(0);
+		setResetGame(false);
+		}
+	}, [resetGame])
+
+
+
+//// fetching the data from an api (via Express) /////
+
+	useEffect(() => {  
+		fetch ("http://localhost:3001/api")
+		.then(res => {
+			return res.json()
+		})
+		.then(totalQuestions => {
+			setQuestions(totalQuestions)
+		})
+	}, [] )
+
+//////////////// RENDERING OF THE APP //////////////////////////
+
+
+//// something to do while there is no data
+
+	if (questions.length === 0) {
+		return (
+			<div className='app'>
+				"loading ..."
+			</div>
+		)
+	}
+
```

Comment:
Super thoughtful here - love how you are thinking about the potential errors or network problems!!